x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/profile/TransportProfileHasPrivilegesAction.java

```diff
@@ -82,6 +88,11 @@ protected void doExecute(Task task, ProfileHasPrivilegesRequest request, ActionL
                 request,
                 ActionListener.wrap(applicationPrivilegeDescriptors -> threadPool.generic().execute(() -> {
                     for (Map.Entry<String, Subject> profileUidToSubject : profileUidAndSubjects) {
+                        // return the partial response if the "has privilege" task got cancelled in the meantime
+                        if (((CancellableTask) task).isCancelled()) {
+                            listener.onResponse(new ProfileHasPrivilegesResponse(hasPrivilegeProfiles, errorProfiles));
+                            return;
+                        }
```

Comment:
I think we could discuss whether to verify the cancellation status before every profile is checked, or every _n_ such checks.
We could also discuss whether an error is more suitable than the partial result.

I don't think the ES JVM node process would notice the performance impovement of every _n_ checks, and partial results is nice and comes for free (there's no headache to implement it).

Reply:
Roger! I'll keep the cancellation status check once every profile, but return the `TaskCancelledException` rather than the partial result.

Reply:
I've pushed [e918fd8](https://github.com/elastic/elasticsearch/pull/87224/commits/e918fd81112d596d1e9d79d71d141769fca4e11a) .